src/swish/http.ms

```diff
@@ -1409,24 +1409,27 @@
   (supervisor:start&link 'http-sup 'one-for-all 0 1
     (http:configure-server #f 0
       (http:url-handler
-       (ws:upgrade conn request me))))
-  (let* ([c (ws:connect "127.0.0.1" (get-http-port) "/" self
-              (ws:options
-               [ping-frequency 10]
-               [pong-timeout 1]))]
-         [_ (receive [#(ws:init ,@c) 'ok])]
-         [s (receive [#(ws:init ,s) s])])
-    (receive (after 20 'ok))
-    (receive (after 4000
-               (let lp ()
-                 (receive (after 10000 'ok)
-                   [,msg
-                    (pretty-print `(,(erlang:now) ,msg))
-                    (lp)]))
-               (throw 'no-client-close))
-      [#(ws:closed ,@c 1002 "no response from ping") 'ok])
-    (receive (after 4000 (throw 'no-server-close))
-      [#(ws:closed ,@s 1002 "no response from ping") 'ok])))
+       (ws:upgrade conn request me
+         (ws:options
+          [ping-frequency 10]
+          [pong-timeout 1])))))
+  (let-values ([(ip op) (connect-tcp "127.0.0.1" (get-http-port))])
+    (for-each
+     (lambda (str)
+       (put-bytevector op (string->utf8 str))
+       (put-bytevector op (string->utf8 "\r\n")))
+     '("GET / HTTP/1.1"
+       "Connection: Upgrade"
+       "Host: 127.0.0.1"
+       "Upgrade: websocket"
+       "Sec-Websocket-Version: 13"
+       "Sec-Websocket-Key: dGhlIHNhbXBsZSBub25jZQ=="))
+    (put-bytevector op (string->utf8 "\r\n"))
+    (flush-output-port op)
+    (let ([s (receive [#(ws:init ,s) s])])
+      (receive (after 20 'ok))
+      (receive (after 1000 (throw 'no-server-close))
+        [#(ws:closed ,@s 1002 "no response from ping") 'ok]))))
```

Comment:
You might add a timeout clause to the receive for ws:init, and I don't think you need the 20 ms wait any more.

Reply:
```
(receive (after 1000 (throw 'no-ws:init))
 [#(ws:init ,s)
  (receive (after 1000 (throw 'no-server-close))
   [#(ws:closed ,@s 1002 "no response from ping") 'ok])])
```

Reply:
You're right. I'm trying a different philosophy on timeouts. For the ws:init case it feels like an assertion for the setup of the whole test -- I'm letting isolate-mat's timeout handle that. For the later timeout, I'm checking behavior so I wanted a specific throw.

I'll remove the 20ms wait. Thanks.

Reply:
I changed my mind. That looks nice without the let binding. I'll take that change too.